From 40ed3cf1338b86734b11e239da7776742b016299 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Yilli=C3=A9?= <yilliee@protonmail.com>
Date: Tue, 21 Jun 2022 08:47:34 +0500
Subject: [PATCH 2/3] Allow bottom-aligning status bar icons

- If the var TW_STATUS_ICONS_BOTTOM_ALIGN is set to true it sets
  the value of row1_header_y such that the bottom of the icons
  corresponds to the bottom of the status bar.
  ( Value calculated by : statusbar height - size of font_m ).
- If it's not set to true, the value of row1_header_y defaults
  to it's old value.

Change-Id: I1b292221f3c34bbac20233595fbc0958d4aa4cc1
---
 Android.mk                      | 4 +++-
 gui/theme/landscape_hdpi/ui.xml | 4 +++-
 gui/theme/landscape_mdpi/ui.xml | 4 +++-
 gui/theme/portrait_hdpi/ui.xml  | 4 +++-
 gui/theme/portrait_mdpi/ui.xml  | 4 +++-
 gui/theme/watch_mdpi/ui.xml     | 4 +++-
 6 files changed, 18 insertions(+), 6 deletions(-)

diff --git a/Android.mk b/Android.mk
index 9727c5f0c..bf1a4b924 100755
--- a/Android.mk
+++ b/Android.mk
@@ -576,7 +576,9 @@ LOCAL_POST_INSTALL_CMD += \
     if [ -n "$(TW_CUSTOM_CLOCK_POS)" ]; then sed -i "s/{clock_pos}/$(TW_CUSTOM_CLOCK_POS)/"  $(TARGET_RECOVERY_ROOT_OUT)/twres/ui.xml; \
     else sed -i "s/{clock_pos}/%center_x%/"  $(TARGET_RECOVERY_ROOT_OUT)/twres/ui.xml; fi; \
     if [ -n "$(TW_CUSTOM_CPU_POS)" ]; then sed -i "s/{cpu_pos}/$(TW_CUSTOM_CPU_POS)/"  $(TARGET_RECOVERY_ROOT_OUT)/twres/ui.xml; \
-    else sed -i "s/{cpu_pos}/%indent%/"  $(TARGET_RECOVERY_ROOT_OUT)/twres/ui.xml; fi;
+    else sed -i "s/{cpu_pos}/%indent%/"  $(TARGET_RECOVERY_ROOT_OUT)/twres/ui.xml; fi; \
+    if [ "$(TW_STATUS_ICONS_BOTTOM_ALIGN)" == "true" ]; then sed -i "s/{statusicons_align}/status_bottomalign_header_y/"  $(TARGET_RECOVERY_ROOT_OUT)/twres/ui.xml; \
+    else sed -i "s/{statusicons_align}/%status_topalign_header_y%/"  $(TARGET_RECOVERY_ROOT_OUT)/twres/ui.xml; fi;
 
 include $(BUILD_EXECUTABLE)
 
diff --git a/gui/theme/landscape_hdpi/ui.xml b/gui/theme/landscape_hdpi/ui.xml
index 2f210f3e8..8960b973e 100755
--- a/gui/theme/landscape_hdpi/ui.xml
+++ b/gui/theme/landscape_hdpi/ui.xml
@@ -119,7 +119,9 @@
 		<variable name="btn4_col2_x_right" value="1218"/>
 		<variable name="btn4_col3_x_right" value="1452"/>
 		<variable name="btn4_col4_x_right" value="1686"/>
-		<variable name="row1_header_y" value="0"/>
+		<variable name="status_topalign_header_y" value="0"/>
+		<variable name="status_bottomalign_header_y" value="16"/>
+		<variable name="row1_header_y" value="{statusicons_alignment}"/>
 		<variable name="row2_header_y" value="48"/>
 		<variable name="row3_header_y" value="60"/>
 		<variable name="row4_header_y" value="126"/>
diff --git a/gui/theme/landscape_mdpi/ui.xml b/gui/theme/landscape_mdpi/ui.xml
index 58a259b2e..cc31494bb 100755
--- a/gui/theme/landscape_mdpi/ui.xml
+++ b/gui/theme/landscape_mdpi/ui.xml
@@ -119,7 +119,9 @@
 		<variable name="btn4_col2_x_right" value="507"/>
 		<variable name="btn4_col3_x_right" value="606"/>
 		<variable name="btn4_col4_x_right" value="703"/>
-		<variable name="row1_header_y" value="0"/>
+		<variable name="status_topalign_header_y" value="0"/>
+		<variable name="status_bottomalign_header_y" value="4"/>
+		<variable name="row1_header_y" value="{statusicons_align}"/>
 		<variable name="row2_header_y" value="20"/>
 		<variable name="row3_header_y" value="24"/>
 		<variable name="row4_header_y" value="52"/>
diff --git a/gui/theme/portrait_hdpi/ui.xml b/gui/theme/portrait_hdpi/ui.xml
index 4eccbadbb..d6e1acba6 100755
--- a/gui/theme/portrait_hdpi/ui.xml
+++ b/gui/theme/portrait_hdpi/ui.xml
@@ -113,7 +113,9 @@
 		<variable name="btn4_col2_x" value="297"/>
 		<variable name="btn4_col3_x" value="558"/>
 		<variable name="btn4_col4_x" value="819"/>
-		<variable name="row1_header_y" value="4"/>
+		<variable name="status_topalign_header_y" value="4"/>
+		<variable name="status_bottomalign_header_y" value="30"/>
+		<variable name="row1_header_y" value="{statusicons_align}"/>
 		<variable name="row2_header_y" value="72"/>
 		<variable name="row3_header_y" value="90"/>
 		<variable name="row4_header_y" value="168"/>
diff --git a/gui/theme/portrait_mdpi/ui.xml b/gui/theme/portrait_mdpi/ui.xml
index 42ead0561..8162d8476 100755
--- a/gui/theme/portrait_mdpi/ui.xml
+++ b/gui/theme/portrait_mdpi/ui.xml
@@ -113,7 +113,9 @@
 		<variable name="btn4_col2_x" value="131"/>
 		<variable name="btn4_col3_x" value="247"/>
 		<variable name="btn4_col4_x" value="363"/>
-		<variable name="row1_header_y" value="2"/>
+		<variable name="status_topalign_header_y" value="2"/>
+		<variable name="status_bottomalign_header_y" value="12"/>
+		<variable name="row1_header_y" value="{statusicons_align}"/>
 		<variable name="row2_header_y" value="30"/>
 		<variable name="row3_header_y" value="40"/>
 		<variable name="row4_header_y" value="74"/>
diff --git a/gui/theme/watch_mdpi/ui.xml b/gui/theme/watch_mdpi/ui.xml
index 478509409..7471421e8 100644
--- a/gui/theme/watch_mdpi/ui.xml
+++ b/gui/theme/watch_mdpi/ui.xml
@@ -96,7 +96,9 @@
 		<variable name="btn4_col2_x" value="86"/>
 		<variable name="btn4_col3_x" value="164"/>
 		<variable name="btn4_col4_x" value="242"/>
-		<variable name="row1_header_y" value="0"/>
+		<variable name="status_topalign_header_y" value="0"/>
+		<variable name="status_bottomalign_header_y" value="6"/>
+		<variable name="row1_header_y" value="{statusicons_align}"/>
 		<variable name="row2_header_y" value="20"/>
 		<variable name="row1_y" value="28"/>
 		<variable name="row1a_y" value="36"/>
-- 
2.36.1

